src/m_pd.h

```diff
@@ -443,8 +465,8 @@ EXTERN int sys_zoomfontheight(int fontsize, int zoom, int worstcase);
 EXTERN int sys_fontwidth(int fontsize);
 EXTERN int sys_fontheight(int fontsize);
 EXTERN void canvas_dataproperties(t_glist *x, t_scalar *sc, t_binbuf *b);
-EXTERN int canvas_open(const t_canvas *x, const char *name, const char *ext,
-    char *dirresult, char **nameresult, unsigned int size, int bin);
+EXTERN bool canvas_open(const t_canvas *x, const char *name, const char *ext,
```

Comment:
please don't use `bool` return types. in general, don't change existing `EXTERN` functions.

Reply:
I've mentioned in some other place that we could keep using `int` as abstract file handles. We could also keep the convention for `-1` to mean failure. The host app only has to map file resources to positive integers, which shouldn't be a problem. (It is actually quite natural if file resources are kept in an array).